src/PointwiseFunctions/AnalyticSolutions/GrMhd/BondiMichelMHD.hpp

```diff
@@ -0,0 +1,296 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <limits>
+#include <pup.h>
+
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Options/Options.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/GeneralRelativity/KerrSchild.hpp"
+#include "PointwiseFunctions/Hydro/EquationsOfState/EquationOfState.hpp"
+#include "PointwiseFunctions/Hydro/Tags.hpp"  // IWYU pragma: keep
+#include "Utilities/MakeArray.hpp"            // IWYU pragma: keep
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+namespace grmhd {
+namespace Solutions {
+
+/*!
+ * \brief Bondi-Michel accretion with superposed magnetic field in Schwarzschild
+ * spacetime.
+ *
+ * Modulo the magnetic field, an analytic solution to the 3-D GrMhd system.
+ * The user specifies the sonic radius \f$r_c\f$ and sonic rest mass density
+ * \f$\rho_c\f$, which are the radius and rest mass density at the sonic point,
+ * the radius at which the fluid's Eulerian velocity as seen by a distant
+ * observer overtakes the local sound speed \f$c_{s,c}^2\f$. With a specified
+ * polytropic exponent \f$\Gamma\f$, these quantities can be related to the
+ * sound speed at infinity \f$c_{s,\inf}^2\f$ using the following relations:
+ *
+ * \f{align*}
+ * c_{s,c}^2 &= \frac{1}{2r_c - 3} \\
+ * c_{s,\inf}^2 &= \Gamma - 1 + (c_{s,c}^2 - \Gamma + 1)\sqrt{1 + 3c_{s,c}^2}
+ * \f}
+ *
+ * In the case of the interstellar medium, the sound
+ * speed is \f$\approx 10^{-4}\f$, which results in a sonic radius of
+ * \f$\approx 10^8 M\f$ for \f$\Gamma \neq 5/3\f$ (Rezzola and Zanotti, 2013).
+ * However, for numerical-testing purposes, it is more common to use a value of
+ * \f$\approx 10 M\f$.
+ *
+ * The density is found via root-finding, through the
+ * Bernoulli equation. As one approaches the sonic radius, a second root makes
+ * an appearance and one must take care to bracket the correct root. This is
+ * done by using the upper bound \f$\frac{\dot{M}}{4\pi}\sqrt{\frac{2}{Mr^3}}\f$
+ *
+ * Additionally specified by the user are the polytropic exponent \f$\Gamma\f$,
+ * and the strength parameter of the magnetic field \f$B\f$.
+ * In Kerr-Schild-Cartesian coordinates \f$(x, y, z)\f$, where
+ * \f$ r = \sqrt{x^2 + y^2 + z^2}\f$, the superposed magnetic field is
+ * \f{align*}
+ * B_x(\vec{x},t) &= \frac{B x}{r^3 \sqrt{1 + 2/r}} \\
+ * B_y(\vec{x},t) &= \frac{B y}{r^3 \sqrt{1 + 2/r}} \\
+ * B_z(\vec{x},t) &= \frac{B z}{r^3 \sqrt{1 + 2/r}}
+ * \f}
+ */
+class BondiMichelMHD {
```

Comment:
Is there a reason to have "MHD" in the class name?  It's in the "grmhd" namespace.